src/events/dns.js

```diff
@@ -0,0 +1,239 @@
+const uuid4 = require('uuid4');
+const dns = require('dns');
+const shimmer = require('shimmer');
+const utils = require('../utils.js');
+const tracer = require('../tracer.js');
+const serverlessEvent = require('../proto/event_pb.js');
+const eventInterface = require('../event.js');
+const errorCode = require('../proto/error_code_pb.js');
+
+// BLACKLIST
+const URL_BLACKLIST = {
+    'tc.epsagon.com': 'endsWith',
+};
+/**
+ * Checks if a URL is in the blacklist
+ * @param {string} url The URL to check
+ * @param {string} path The Path to check (optional)
+ * @returns {boolean} True if it is in the blacklist, False otherwise
+ */
+function isBlacklistURL(url, path) {
+    return Object.keys(URL_BLACKLIST).some((key) => {
+        if (typeof URL_BLACKLIST[key] === typeof (() => {})) {
+            return URL_BLACKLIST[key](url, key, path);
+        }
+        return url[URL_BLACKLIST[key]](key);
+    });
+}
+
+const rrtypesMethods = {
+    resolveAny: 'ANY',
+    resolve6: 'AAAA',
+    resolve4: 'A',
+    resolveCname: 'CNAME',
+    resolveMx: 'MX',
+    resolveNaptr: 'NAPTR',
+    resolveNs: 'NS',
+    resolvePtr: 'PTR',
+    resolveSoa: 'SOA',
+    resolveSrv: 'SRV',
+    resolveTxt: 'TXT',
+};
+
+const initialDnsEvent = (name) => {
+    const startTime = Date.now();
+    const resource = new serverlessEvent.Resource([
+        'dns',
+        'dns',
+        name,
+    ]);
+    const dnsEvent = new serverlessEvent.Event([
+        `dns-${uuid4()}`,
+        utils.createTimestampFromTime(startTime),
+        null,
+        'dns',
+        0,
+        errorCode.ErrorCode.OK,
+    ]);
+    dnsEvent.setResource(resource);
+    return { dnsEvent, startTime };
+};
+
+const endDurationAndAddCallbackMetadata = (dnsEvent, startTime, err, metadata) => {
+    dnsEvent.setDuration(utils.createDurationTimestamp(startTime));
+    if (err) {
+        eventInterface.setException(dnsEvent, err);
+    } else {
+        eventInterface.addToMetadata(dnsEvent, { ...metadata });
```

Comment:
are you sure you want to add the entire metadata? maybe do it only when metadataOnly=False (third argument instead of second)? consult with @henprtz

Reply:
also no need to re-spread metadata i guess?

Reply:
also make sure the naming and size of the response is reasonable. whitelisting what we add to the metadata is usually the better approach

Reply:
> are you sure you want to add the entire metadata? maybe do it only when metadataOnly=False (third argument instead of second)? consult with @henprtz

At a minimum (and most commonly) is a string/integer and at maximum it could be an array/object of strings (https://nodejs.org/api/dns.html). I think it's relevant data for those who decide to trace dns.
example for the longest metadata I got so far:
```
{
"retry":3600
"expire":604800
"minttl":3601
"nsname": "dns1.registrar-servers.com" 
"serial":1573449957
"refresh":43200
"hostmaster": "hostmaster.registrar-servers.com" 
}
```
Hostname: e


> also no need to re-spread metadata i guess?

It's not necessary after a small change i made in the code, tnx.



> also make sure the naming and size of the response is reasonable. whitelisting what we add to the metadata is usually the better approach

@galbash There is any size limitation of the metadata i need to know?